core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -5654,21 +5657,165 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
                 switch (opcode) {
                     /* Memory */
                     case SIMD_v128_load:
+                    {
+                        uint32 offset, addr;
+                        offset =
+                            read_uint32(frame_ip); // TODO Check with an offset!
+                        addr = GET_OPERAND(uint32, I32, 0);
+                        frame_ip += 2;
+                        addr_ret = GET_OFFSET();
+                        CHECK_MEMORY_OVERFLOW(16);
+                        PUT_V128_TO_ADDR(frame_lp + addr_ret, LOAD_V128(maddr));
+                        break;
+                    }
                     case SIMD_v128_load8x8_s:
+#if ENABLE_NEON == 1
+                    {
+                        uint32 offset, addr;
+                        offset = read_uint32(frame_ip);
+                        addr = GET_OPERAND(uint32, I32, 0);
+                        frame_ip += 2;
+                        addr_ret = GET_OFFSET();
+                        CHECK_MEMORY_OVERFLOW(16);
+
+                        int8x8_t v8 = vld1_s8((const int8_t *)maddr);
+                        v8 = vrev64_s8(v8);
+                        int16x8_t v16 = vmovl_s8(v8);
```

Comment:
Had better not use arch/os specified API here, could we wrap them to common APIs? And are there common implementations which doesn't depend on specified arch/os?

Reply:
+1

using arch specific things directly like this PR should be the last resort after considering other options.

have you already considered libraries like simde? or compiler intrinsics? or even pure C implementation?
cf. https://github.com/yamt/toywasm/blob/e9226fae3e610d37742aa7afc8b0b7a7cf04f941/lib/insn_impl_simd.h#L8-L12

Reply:
Happy to use simde - I'll update this PR adding that and implementing a couple of opcodes.

Reply:
Done, added the library and implemented the swizzle opcode using it. Should be quite quick to add the rest of them now.